### ISSUE TITLE
Enhance the Kaniko builder

### ISSUE THREAD

**Issue by r2d4** [GoogleContainerTools/skaffold#671]
Namely

- [x] Surface some of the flags that have been added since the initial release
- [x] The option to use an existing cluster secret instead of creating one from a keyfile passed in.
- [x] Generate unique pod names
- [x] Configurable namespace for kaniko builder

cc @sharifelgamal @priyawadhwa

**Comment by JonnyBGod** [comment]
Also would be good to be able to use Kaniko by passing just the local context and destination skipping the need for a GC Bucket

**Comment by dlorenc** [comment]
A few other improvements:

- Support a configurable namespace for kaniko
- Improve secret management (specifying existing secrets, etc.)
- Generate unique names for the pod (in case of errors with earlier versions, etc.)

**Comment by dlorenc** [comment]
A few other things left to do. Updated in the top comment to track in one spot.

**Comment by dlorenc** [comment]
> Also would be good to be able to use Kaniko by passing just the local context and destination skipping the need for a GC Bucket

I think that's more of a kaniko limitation than a skaffold issue - would you mind opening an issue in that repo with the use case?

**Comment by priyawadhwa** [comment]
You can pass in a local directory as the build context to kaniko with the `--context` flag. I think skaffold would just have to mount the local context into the pod to be able to use it.

**Comment by dlorenc** [comment]
I think the problem would be first getting the source from the local machine into the cluster and onto the node,although it's true skaffold might be able to do that before starting the kaniko container.

**Comment by r2d4** [comment]
Closing this with #803, which upgrades the version to v0.2.0. We can track more specific feature requests in another issue.
